src/Spreadsheet.tsx

```diff
@@ -235,13 +236,12 @@ const Spreadsheet = <CellType extends Types.CellBase>(
   // Listen to data changes
   const prevDataRef = React.useRef<Matrix.Matrix<CellType>>(state.model.data);
   React.useEffect(() => {
-    if (state.model.data !== prevDataRef.current) {
+    if (!deepArrayComparison(state.model.data, prevDataRef.current)) {
       // Call on change only if the data change internal
-      if (state.model.data !== props.data) {
+      if (!deepArrayComparison(state.model.data, props.data)) {
```

Comment:
I'm sorry but although this is a working solution for small datasets, for big ones it could be very problematic. The fix should be to make sure the reference isn't different in the first place.